html/pages/addhost.inc.php

```diff
@@ -10,7 +11,11 @@
     exit;
 }
 
-$hostname = isset($_POST['hostname']) ? clean($_POST['hostname']) : false;
+$hostname = (isset($_POST['hostname']) && clean($_POST['hostname']) != "") ? clean($_POST['hostname']) : false;
```

Comment:
Can you change this from a ternary to a regular if else?  That would make it more readable.